astropy/conftest.py

```diff
@@ -20,22 +20,20 @@
 
 from astropy import __version__
 from astropy.tests.helper import enable_deprecations_as_exceptions
+from astropy.utils.compat.optional_deps import HAS_MATPLOTLIB
 
-try:
-    # This is needed to silence a warning from matplotlib caused by
-    # PyInstaller's matplotlib runtime hook.  This can be removed once the
-    # issue is fixed upstream in PyInstaller, and only impacts us when running
-    # the tests from a PyInstaller bundle.
-    # See https://github.com/astropy/astropy/issues/10785
-    if getattr(sys, 'frozen', False) and hasattr(sys, '_MEIPASS'):
-        # The above checks whether we are running in a PyInstaller bundle.
-        warnings.filterwarnings("ignore", "(?s).*MATPLOTLIBDATA.*",
-                                category=UserWarning)
+if HAS_MATPLOTLIB:
     import matplotlib
-except ImportError:
-    HAS_MATPLOTLIB = False
-else:
-    HAS_MATPLOTLIB = True
+
+# This is needed to silence a warning from matplotlib caused by
+# PyInstaller's matplotlib runtime hook.  This can be removed once the
+# issue is fixed upstream in PyInstaller, and only impacts us when running
+# the tests from a PyInstaller bundle.
+# See https://github.com/astropy/astropy/issues/10785
+if getattr(sys, 'frozen', False) and hasattr(sys, '_MEIPASS'):
+    # The above checks whether we are running in a PyInstaller bundle.
+    warnings.filterwarnings("ignore", "(?s).*MATPLOTLIBDATA.*",
```

Comment:
Maybe this broken `pyinstaller` job...

Reply:
Is the `pyinstaller` job working on master? I guess I can indent this block to inside `if HAS_MATPLOTLIB` but I don't immediately see how these changes could be causing a new error

Reply:
Yes, it runs nightly. I restarted the one from last night and it still passes on `master`.

Reply:
Oh, I see, the warning from #10785 isn't getting silenced...hmm ok looking into it

Reply:
Oops, I reversed the order - the filterwarnings needs to go before the import!